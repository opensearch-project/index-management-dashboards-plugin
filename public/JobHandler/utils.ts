/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
export const EVENT_MAP = {
  REINDEX_COMPLETE: "REINDEX_COMPLETE",
  SPLIT_COMPLETE: "SPLIT_COMPLETE",
  SHRINK_COMPLETE: "SHRINK_COMPLETE",
};

export const triggerEvent = (eventName: string, data?: unknown) => {
  const event = new CustomEvent(eventName, {
    detail: data,
  });
  window.dispatchEvent(event);
};

export const listenEvent = (eventName: string, callback: () => void) => {
  window.addEventListener(eventName, callback);
};

export const destroyListener = (eventName: string, callback: () => void) => {
  window.removeEventListener(eventName, callback);
};
