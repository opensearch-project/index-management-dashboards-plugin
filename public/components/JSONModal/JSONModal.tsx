/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from "react";
import {
  EuiModal,
  EuiModalBody,
  EuiModalFooter,
  EuiModalHeader,
  EuiModalHeaderTitle,
  EuiOverlayMask,
  EuiCodeBlock,
  EuiButtonEmpty,
} from "@elastic/eui";

interface JSONModalProps {
  title: string;
  json: object;
  onClose: () => void;
}

const JSONModal: React.SFC<JSONModalProps> = ({ title, json, onClose }) => {
  return (
    <EuiOverlayMask>
      <EuiModal onClose={onClose} style={{ padding: "5px 30px" }}>
        <EuiModalHeader>
          <EuiModalHeaderTitle>{title}</EuiModalHeaderTitle>
        </EuiModalHeader>

        <EuiModalBody>
          <EuiCodeBlock language="json" fontSize="m" paddingSize="m" overflowHeight={600} inline={false} isCopyable>
            {JSON.stringify(json, null, 4)}
          </EuiCodeBlock>
        </EuiModalBody>

        <EuiModalFooter>
          <EuiButtonEmpty onClick={onClose} data-test-subj="jsonModalCloseButton">
            Close
          </EuiButtonEmpty>
        </EuiModalFooter>
      </EuiModal>
    </EuiOverlayMask>
  );
};

export default JSONModal;
