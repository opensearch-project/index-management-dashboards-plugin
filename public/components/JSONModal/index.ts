/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import JSONModal from "./JSONModal";

export default JSONModal;
