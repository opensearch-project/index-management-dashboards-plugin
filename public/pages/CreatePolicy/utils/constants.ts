/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export const DEFAULT_POLICY = JSON.stringify(
  {
    policy: {
      description: "A simple default policy that changes the replica count between hot and cold states.",
      default_state: "example_hot_state",
      states: [
        {
          name: "example_hot_state",
          actions: [{ replica_count: { number_of_replicas: 5 } }],
          transitions: [{ state_name: "example_cold_state", conditions: { min_index_age: "30d" } }],
        },
        {
          name: "example_cold_state",
          actions: [{ replica_count: { number_of_replicas: 2 } }],
          transitions: [],
        },
      ],
      ism_template: {
        index_patterns: ["example-index-*"],
      },
    },
  },
  null,
  4
);
