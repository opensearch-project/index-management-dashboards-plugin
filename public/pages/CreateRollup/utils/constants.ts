/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export const EMPTY_ROLLUP = JSON.stringify({
  rollup: {
    continuous: false,
    description: "",
    dimensions: [
      {
        date_histogram: {
          source_field: "",
          fixed_interval: "1h",
          timezone: "UTC",
        },
      },
    ],
    enabled: true,
    metrics: [],
    page_size: 1000,
    roles: [],
    schedule: {
      interval: {
        start_time: 234802,
        period: 1,
        unit: "MINUTES",
      },
    },
    source_index: "",
    target_index: "",
  },
});

export const DelayTimeunitOptions = [
  { value: "SECONDS", text: "Second(s)" },
  { value: "MINUTES", text: "Minute(s)" },
  { value: "HOURS", text: "Hour(s)" },
  { value: "DAYS", text: "Day(s)" },
];

export const ScheduleIntervalTimeunitOptions = [
  { value: "MINUTES", text: "Minute(s)" },
  { value: "HOURS", text: "Hour(s)" },
  { value: "DAYS", text: "Day(s)" },
];

export const AddFieldsColumns = [
  {
    field: "label",
    name: "Field name",
    sortable: true,
  },
  {
    field: "type",
    name: "Field type",
    sortable: true,
    render: (type: string | undefined) => (type == null || type === undefined ? "-" : type),
  },
];

export const WHERE_BOOLEAN_FILTERS = [
  { text: "Select value", value: "" },
  { text: "True", value: true },
  { text: "False", value: false },
];

export const OPERATORS_MAP = {
  IS: "is",
  IS_NOT: "is_not",
  IS_NULL: "is_null",
  IS_NOT_NULL: "is_not_null",
  IS_GREATER: "is_greater",
  IS_GREATER_EQUAL: "is_greater_equal",
  IS_LESS: "is_less",
  IS_LESS_EQUAL: "is_less_equal",
  STARTS_WITH: "starts_with",
  ENDS_WITH: "ends_with",
  CONTAINS: "contains",
  NOT_CONTAINS: "does_not_contains",
  IN_RANGE: "in_range",
  NOT_IN_RANGE: "not_in_range",
};

export const COMPARISON_OPERATORS = [
  { text: "is", value: OPERATORS_MAP.IS, dataTypes: ["number", "text", "keyword", "boolean"] },
  {
    text: "is not",
    value: OPERATORS_MAP.IS_NOT,
    dataTypes: ["number", "text", "keyword", "boolean"],
  },
  {
    text: "is null",
    value: OPERATORS_MAP.IS_NULL,
    dataTypes: ["number", "text", "keyword", "boolean"],
  },
  {
    text: "is not null",
    value: OPERATORS_MAP.IS_NOT_NULL,
    dataTypes: ["number", "text", "keyword"],
  },
  { text: "is greater than", value: OPERATORS_MAP.IS_GREATER, dataTypes: ["number"] },
  { text: "is greater than equal", value: OPERATORS_MAP.IS_GREATER_EQUAL, dataTypes: ["number"] },
  { text: "is less than", value: OPERATORS_MAP.IS_LESS, dataTypes: ["number"] },
  { text: "is less than equal", value: OPERATORS_MAP.IS_LESS_EQUAL, dataTypes: ["number"] },
  { text: "is in range", value: OPERATORS_MAP.IN_RANGE, dataTypes: ["number"] },
  { text: "is not in range", value: OPERATORS_MAP.NOT_IN_RANGE, dataTypes: ["number"] },
  { text: "starts with", value: OPERATORS_MAP.STARTS_WITH, dataTypes: ["text", "keyword"] },
  { text: "ends with", value: OPERATORS_MAP.ENDS_WITH, dataTypes: ["text", "keyword"] },
  { text: "contains", value: OPERATORS_MAP.CONTAINS, dataTypes: ["text", "keyword"] },
  { text: "does not contains", value: OPERATORS_MAP.NOT_CONTAINS, dataTypes: ["text"] },
];
