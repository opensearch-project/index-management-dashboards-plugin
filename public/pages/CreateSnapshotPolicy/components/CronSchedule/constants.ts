/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import moment from "moment-timezone";

export const CRON_SCHEDULE_FREQUENCY_TYPE = [
  { value: "hourly", text: "Hourly" },
  { value: "daily", text: "Daily" },
  { value: "weekly", text: "Weekly" },
  { value: "monthly", text: "Monthly" },
  { value: "custom", text: "Custom (Cron expression)" },
];

export const TIMEZONES = moment.tz.names().map((tz) => ({ label: tz, text: tz }));

export const WEEK_DAYS = ["MON", "TUE", "WED", "THU", "FRI", "SAT", "SUN"];
