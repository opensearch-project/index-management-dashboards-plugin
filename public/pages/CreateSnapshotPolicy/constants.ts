/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { SMPolicy } from "../../../models/interfaces";

/**
 * Every time Component init we want to give a different default object
 */
export const getDefaultSMPolicy = (): SMPolicy => ({
  name: "",
  description: "",
  enabled: true,
  creation: {
    schedule: {
      cron: {
        expression: "0 20 * * *",
        timezone: "UTC",
      },
    },
  },
  snapshot_config: {
    repository: "",
    // ignore_unavailable: false,
    // include_global_state: false,
    // partial: false,
    // date_expression: "yyyy-MM-dd-HH:mm",
  },
});

export const maxAgeUnitOptions = [
  { value: "d", text: "Days" },
  { value: "h", text: "Hours" },
];

export const DEFAULT_INDEX_OPTIONS = [{ label: "*" }];

export const ERROR_PROMPT = {
  NAME: "Name must be provided.",
  REPO: "Repository must be provided.",
  TIMEZONE: "Time zone must be provided.",
};

export const DEFAULT_DATE_FORMAT = "yyyy-MM-dd'T'HH:mm:ss";
export const DEFAULT_DATE_FORMAT_TIMEZONE = "UTC";
