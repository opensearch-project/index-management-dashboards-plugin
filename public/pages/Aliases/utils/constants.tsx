/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import { SortDirection } from "../../../utils/constants";
import { IAlias } from "../interface";

export const DEFAULT_PAGE_SIZE_OPTIONS = [5, 10, 20, 50];

export const DEFAULT_QUERY_PARAMS = {
  from: "0",
  size: "20",
  search: "",
  sortField: "alias" as keyof IAlias,
  sortDirection: SortDirection.DESC,
  status: "",
  dataSourceId: "",
};
