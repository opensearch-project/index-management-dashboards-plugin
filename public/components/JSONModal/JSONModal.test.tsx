/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, fireEvent } from "@testing-library/react";
import JSONModal from "./JSONModal";
import { DEFAULT_LEGACY_ERROR_NOTIFICATION } from "../../pages/VisualCreatePolicy/utils/constants";

describe("<JSONModal /> spec", () => {
  it("renders the component", () => {
    render(
      <JSONModal
        title="Some modal title"
        json={DEFAULT_LEGACY_ERROR_NOTIFICATION} // just some random json
        onClose={() => {}}
      />
    );
    // EuiOverlayMask appends an element to the body so we should have two, an empty div from react-test-library
    // and our EuiOverlayMask element
    expect(document.body.children).toHaveLength(2);
    expect(document.body.children[1]).toMatchSnapshot();
  });

  it("calls close when close button clicked", () => {
    const onClose = jest.fn();
    const { getByTestId } = render(
      <JSONModal
        title="Some modal title"
        json={DEFAULT_LEGACY_ERROR_NOTIFICATION} // just some random json
        onClose={onClose}
      />
    );

    fireEvent.click(getByTestId("jsonModalCloseButton"));
    expect(onClose).toHaveBeenCalled();
  });
});
