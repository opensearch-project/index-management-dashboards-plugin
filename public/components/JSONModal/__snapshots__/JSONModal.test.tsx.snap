// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JSONModal /> spec renders the component 1`] = `
<div
  class="euiOverlayMask euiOverlayMask--aboveHeader"
>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="1"
  />
  <div
    data-focus-lock-disabled="false"
  >
    <div
      class="euiModal euiModal--maxWidth-default"
      style="padding: 5px 30px;"
      tabindex="0"
    >
      <button
        aria-label="Closes this modal window"
        class="euiButtonIcon euiButtonIcon--text euiModal__closeIcon"
        type="button"
      >
        EuiIconMock
      </button>
      <div
        class="euiModal__flex"
      >
        <div
          class="euiModalHeader"
        >
          <div
            class="euiModalHeader__title"
          >
            Some modal title
          </div>
        </div>
        <div
          class="euiModalBody"
        >
          <div
            class="euiModalBody__overflow"
          >
            <div
              class="euiCodeBlock euiCodeBlock--fontMedium euiCodeBlock--paddingMedium euiCodeBlock--hasControls"
              style="max-height: 600px;"
            >
              <pre
                class="euiCodeBlock__pre euiCodeBlock__pre--whiteSpacePreWrap"
                style="max-height: 600px;"
              >
                <code
                  class="euiCodeBlock__code json hljs"
                >
                  {
    
                  <span
                    class="hljs-attr"
                  >
                    "destination"
                  </span>
                  : {
        
                  <span
                    class="hljs-attr"
                  >
                    "slack"
                  </span>
                  : {
            
                  <span
                    class="hljs-attr"
                  >
                    "url"
                  </span>
                  : 
                  <span
                    class="hljs-string"
                  >
                    "&lt;url&gt;"
                  </span>
                  
        }
    },
    
                  <span
                    class="hljs-attr"
                  >
                    "message_template"
                  </span>
                  : {
        
                  <span
                    class="hljs-attr"
                  >
                    "source"
                  </span>
                  : 
                  <span
                    class="hljs-string"
                  >
                    "The index {{ctx.index}} failed during policy execution."
                  </span>
                  
    }
}
                </code>
              </pre>
              <div
                class="euiCodeBlock__controls"
              >
                <button
                  aria-label="Expand"
                  class="euiButtonIcon euiButtonIcon--text euiCodeBlock__fullScreenButton"
                  type="button"
                >
                  EuiIconMock
                </button>
                <div
                  class="euiCodeBlock__copyButton"
                >
                  <span
                    class="euiToolTipAnchor"
                  >
                    <button
                      aria-label="Copy"
                      class="euiButtonIcon euiButtonIcon--text"
                      type="button"
                    >
                      EuiIconMock
                    </button>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="euiModalFooter"
        >
          <button
            class="euiButtonEmpty euiButtonEmpty--primary"
            data-test-subj="jsonModalCloseButton"
            type="button"
          >
            <span
              class="euiButtonContent euiButtonEmpty__content"
            >
              <span
                class="euiButtonEmpty__text"
              >
                Close
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
</div>
`;
